src/sailing_robot/scripts/wind_direction_average

```diff
@@ -0,0 +1,52 @@
+#!/usr/bin/python
+
+import math
+import collections
+import rospy
+from std_msgs.msg import Float32, Float64
+
+
+def update_heading(msg):
+    heading = msg.data
```

Comment:
This does nothing as things stand, because `heading` is a local variable inside the function - it will be set and immediately discarded when the function exits.

The best way to approach this is to make a class that holds the data, and then the update callbacks are methods on the class that set e.g. `self.heading = msg.data`.

(It could also be done by using global variables, but that's generally considered bad practice)